main.tf

```diff
@@ -35,22 +37,29 @@ resource "aws_security_group" "this" {
     cidr_blocks = ["0.0.0.0/0"]
   }
 
-  tags = var.tags
+  tags = merge(var.tags, {Name = "vpc all interface endpoints", env = terraform.workspace})
+}
+
+resource "aws_security_group" "all" {
```

Comment:
what do you think about removing this default `ingress/egress` as well?

Reply:
Ahh, I see. _That's_ what the new resource is really providing, a shell in which rules can be added.

Ok, so that makes things interesting. The rationale for using inline ingress/egress rules is that terraform will enforce them _exclusively_. E.g. if outside terraform any SG rules are added, manually perhaps, terraform will detect that as a difference and generate a plan that restores the rules to exactly what is defined in the config.

Using separate `aws_security_group_rule` resources, that exclusivity is not enforced.

I'd like to maintain the option for exclusive inline rules, but that does make things more complex. Can probably get there using [dynamic blocks](https://www.terraform.io/docs/configuration/expressions.html#dynamic-blocks), perhaps with another conditional variable.

Would you be up for trying that approach?

Reply:
Great idea yourself.

Interesting with using inline security group rules to keep an audit trail. I've generally avoided this approach for a couple reasons: the diffs can be a pain to go through when implementing changes and it also doesn't allow for passing the sg for other modules to modify, making it easy to create dependency chains.

Hopefully, I've resolved your issues around this.